airframe-surface/shared/src/main/scala/wvlet/airframe/surface/SurfaceMacros.scala

```diff
@@ -571,6 +590,21 @@ private[surface] object SurfaceMacros {
         q"wvlet.airframe.surface.ExistentialType"
       case t @ RefinedType(List(_, baseType), decl) =>
         newGenericSurfaceOf(baseType)
+      case t if hasStringUnapply(t) =>
+        t.typeSymbol.fullName.split("\\.").toList match {
+          case p1 :: tl =>
+            // companion object package -> Select(Select( .... ))) tree
+            val ex = tl.foldLeft[Tree](Ident(TermName(p1))) { (z, n) =>
+              Select(z, TermName(n))
+            }
```

Comment:
This generates a tree like `Select(Select(Ident(TermName(java)), TermName(obj)), TermName(Object))` for `java.lang.Object` and it selects a companion object. I see...

Reply:
There might be a better way to produce this expression, but I could not find it...